config/config.go

```diff
@@ -1042,8 +1042,21 @@ func (cfg *ConsensusConfig) Precommit(round int32) time.Duration {
 
 // Commit returns the amount of time to wait for straggler votes after receiving +2/3 precommits
 // for a single block (ie. a commit).
-func (cfg *ConsensusConfig) Commit(t time.Time) time.Time {
-	return t.Add(cfg.TimeoutCommit)
+func (cfg *ConsensusConfig) Commit(t time.Time, blockSizeBytes int64) time.Time {
+	// adjust the timeout based on how large the block size is for a total of
+	// 8MB which would decrease the timeout commit by roughly 4 seconds
+	perByteDelay := time.Duration(time.Microsecond / 2)
+	maxBlockSize := int64(8 * 1024 * 1024)
+	if blockSizeBytes > maxBlockSize {
+		blockSizeBytes = maxBlockSize
+	}
+
+	timeoutCommit := cfg.TimeoutCommit - (time.Duration(blockSizeBytes) * perByteDelay)
```

Comment:
what would be the best of determining this value? I guess adjusting on mocha?

would you describe the upside being simplicity and the downside being that changes to the network require re-calibrating this value?

Reply:
Yes, this function should be determined empirically by running 100 node networks with varying block sizes from 0MB to 8MB and modelling that into a function.

> would you describe the upside being simplicity and the downside being that changes to the network require re-calibrating this value?

Yes it's a good simple temporary solution. If we go down the road of either compact blocks or proposer based timestamps. Those solutions can replace this current one